### ISSUE TITLE
find a bug about train on giou_loss

### ISSUE THREAD

**Issue by cxf2015** [dlyldxwl/Stronger-One-stage-detector-with-much-Tricks#1]
in box_uils.py:
decode() function 
boxes[:, :2] -= boxes[:, 2:] / 2
    boxes[:, 2:] += boxes[:, :2]
have inplace operation error.
should be change to:
boxes_temp1=boxes[:,:2]-(boxes[:, 2:] / 2)
boxes_temp2=boxes[:, 2:]+boxes[:, :2]
boxes_out=torch.cat((boxes_temp1,boxes_temp2),1)

**Comment by dlyldxwl** [comment]
@cxf2015 
This inplace operation is correct.
The result of GIOU with SSD is worse,  only 78+. 
How about you ?

**Comment by dlyldxwl** [comment]
About GIOU, I update the loss.py and detection.py files ( for bbox after decode, apply min and max function to get the left-up and right-down points). 
Furthermore, update the best result, 80.58.
